vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java

```diff
@@ -65,14 +68,28 @@ public MailClient sendMail(MailMessage message, Handler<AsyncResult<MailResult>>
     Context context = vertx.getOrCreateContext();
     if (!closed) {
       if (validateHeaders(message, resultHandler, context)) {
-        connectionPool.getConnection(result -> {
-          if (result.succeeded()) {
-            result.result().setErrorHandler(th -> handleError(th, resultHandler, context));
-            sendMessage(message, result.result(), resultHandler, context);
-          } else {
-            handleError(result.cause(), resultHandler, context);
-          }
-        });
+        vertx.executeBlocking(
+            fut -> {
+              if (hostname == null) {
```

Comment:
Actually the hostname should not be set in the executeBlocking block but instead resolve the Future, so it should be something like:
```
if (hostname == null) {
  vertx.<String>executeBlocking(
  fut -> ... fut.complete(value_of_hostname),
  ar -> {
    if (ar.succeeded()) {
      hostName = ar.result();
    } else { ... }
  };)
}
```